Document/0x05b-Basic-Security_Testing.md

```diff
@@ -247,20 +248,22 @@ You'll come across different use cases on how you can use adb commands when test
 
 ##### Angr
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1235>
+Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. In other words: given a binary and a requested state, Angr will try to get to that to the requested state in the binary, using formal methods to find a path, as well as brute forcing. This can make reverse-engineerig "I just want a solution style" - way easier than with for instance IDA Pro, Radare2, or Ghidra. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with _native_ Android binaries.
 
-Angr is a Python framework for analyzing binaries. It is useful for both static and dynamic symbolic ("concolic") analysis. Angr operates on the VEX intermediate language and comes with a loader for ELF/ARM binaries, so it is perfect for dealing with native Android binaries.
+Angr allows for disassempbly, program instrumentation, symbolic execution, control-flow analysis, data-dependency analysis, decompilation and more, given a large set of plugins.
 
 Since version 8 Angr is based on Python 3, and it's available from PyPI. With pip, it's easy to install on \*nix operating systems and Mac OS:
 
 ```shell
-$ pip install angr
+$ mkvirtualenv --python=$(which python3) angr && python -m pip install angr
 ```
 
-Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else.
+Creating a dedicated virtual environment with Virtualenv is recommended because some of its dependencies contain forked versions Z3 and PyVEX, which overwrite the original versions. You can skip this step if you don't use these libraries for anything else. Alternatively, there is always a docker container which you can use.
 
 Comprehensive documentation, including an installation guide, tutorials, and usage examples is available on [Gitbooks page of angr](https://docs.angr.io/ "angr"). A complete [API reference](https://angr.io/api-doc/ "angr API") is also available.
 
+You can use angr with a pythorn REPL - such as iPython - or script your approaches. Although angr has a bit of a steep learning curve, we do recommend using it when you want to brute force your way to a given state of an executable. See [Binary Analysis Frameworks](###Binary Analysis Frameworks) as a great example of how this can work.
```

Comment:
Fixes typo, section name (actually is Binary Analysis now, but angr is in Symbolic Execution), also referencing with several #'s doesn't work. only 1 is required

```suggestion
You can use angr from a Python REPL - such as iPython - or script your approaches. Although angr has a bit of a steep learning curve, we do recommend using it when you want to brute force your way to a given state of an executable. See section [Symbolic Execution](#Symbolic-Execution) as a great example on how this can work.
```